contracts/StablecoinConverter.sol

```diff
@@ -224,23 +226,81 @@ contract StablecoinConverter is EpochTokenLocker {
         tokenConservation.checkTokenConservation();
         documentTrades(batchIndex, owners, orderIds, volumes, tokenIdsForPrice);
     }
+    /**
+     * Public View Methods
+     */
+
+    /** @dev View returning ID of listed tokens
+      * @param addr address of listed token.
+      * @return tokenId as stored, via BiMap, within the contract.
+      */
+    function tokenAddressToIdMap(address addr) public view returns (uint16) {
+        return IdToAddressBiMap.getId(registeredTokens, addr);
+    }
+
+    /** @dev View returning address of listed token by ID
+      * @param id tokenId as stored, via BiMap, within the contract.
+      * @return address of (listed) token
+      */
+    function tokenIdToAddressMap(uint16 id) public view returns (address) {
+        return IdToAddressBiMap.getAddressAt(registeredTokens, id);
+    }
 
+    /** @dev View returning all currently stored, byte-encoded sell orders
+      * @return encoded bytes representing all orders ordered by (user, index)
+      */
+    function getEncodedAuctionElements() public view returns (bytes memory elements) {
+        if (allUsers.size() > 0) {
+            address user = allUsers.first();
+            bool stop = false;
+            while (!stop) {
+                for (uint i = 0; i < orders[user].length; i++) {
+                    Order memory order = orders[user][i];
+                    elements = elements.concat(encodeAuctionElement(
+                        user,
+                        getBalance(user, tokenIdToAddressMap(order.sellToken)),
+                        order
+                    ));
+                }
+                if (user == allUsers.last) {
+                    stop = true;
+                } else {
+                    user = allUsers.next(user);
+                }
+            }
+        }
+        return elements;
+    }
+
+    /** @dev called by solvers to determine their solution better than the current winner.
```

Comment:
I think there is an *if* missing in this sentence. Also not sure if we should make assumptions on why someone would call this method (there might be many other reasons).

Reply:
```suggestion
    /** @dev gets the objective value of currently winning solution.
```